File: Tests/Scraping/LepersColonyPageScrapingTests.m
//  LepersColonyPageScrapingTests.m
//
//  Copyright 2013 Awful Contributors. CC BY-NC-SA 3.0 US https://github.com/Awful/Awful.app

#import "AwfulScrapingTestCase.h"
#import "AwfulLepersColonyPageScraper.h"

@interface LepersColonyPageScrapingTests : AwfulScrapingTestCase

@end

@implementation LepersColonyPageScrapingTests

+ (Class)scraperClass
{
    return [AwfulLepersColonyPageScraper class];
}

- (void)testFirstPage
{
    AwfulLepersColonyPageScraper *scraper = [self scrapeFixtureNamed:@"banlist"];
    NSArray *bans = scraper.bans;
    XCTAssertEqual(bans.count, (NSUInteger)50);
    NSArray *allUsers = [AwfulUser fetchAllInManagedObjectContext:self.managedObjectContext];
    XCTAssertEqual(allUsers.count, (NSUInteger)71);
    NSArray *allPosts = [AwfulPost fetchAllInManagedObjectContext:self.managedObjectContext];
    XCTAssertEqual(allPosts.count, (NSUInteger)46);
    
    AwfulBan *first = bans[0];
    XCTAssertEqual(first.punishment, AwfulPunishmentProbation);
    XCTAssertEqualObjects(first.post.postID, @"421665753");
    XCTAssertEqual(first.date.timeIntervalSince1970, 1384078200.);
    XCTAssertEqualObjects(first.user.username, @"Kheldragar");
    XCTAssertEqualObjects(first.user.userID, @"202925");
    XCTAssertNotEqual([first.reasonHTML rangeOfString:@"shitty as you"].location, (NSUInteger)NSNotFound);
    XCTAssertEqualObjects(first.requester.username, @"Ralp");
    XCTAssertEqualObjects(first.requester.userID, @"61644");
    XCTAssertEqualObjects(first.approver, first.requester);
}

@end

File: Tests/Scraping/LepersColonyPageScrapingTests.swift
//  LepersColonyPageScrapingTests.swift
//
//  Copyright 2013 Awful Contributors. CC BY-NC-SA 3.0 US https://github.com/Awful/Awful.app

import XCTest
import Awful

class LepersColonyPageScrapingTests: ScrapingTestCase {
    
    override class func scraperClass() -> AnyClass {
        return LepersColonyPageScraper.self
    }
    
    func testFirstPage() {
        let scraper = scrapeFixtureNamed("banlist") as LepersColonyPageScraper
        let punishments = scraper.punishments as [Punishment]
        XCTAssertTrue(punishments.count == 50)
        XCTAssertTrue(User.numberOfObjectsInManagedObjectContext(managedObjectContext) == 71)
        XCTAssertTrue(Post.numberOfObjectsInManagedObjectContext(managedObjectContext) == 46)
        
        let first = punishments[0]
        XCTAssertEqual(first.sentence, PunishmentSentence.Probation)
        XCTAssertEqual(first.post!.postID, "421665753")
        XCTAssertEqual(first.date.timeIntervalSince1970, 1384078200)
        XCTAssertEqual(first.subject.username!, "Kheldragar")
        XCTAssertEqual(first.subject.userID!, "202925")
        XCTAssertTrue(first.reasonHTML!.rangeOfString("shitty as you") != nil)
        XCTAssertEqual(first.requester!.username!, "Ralp")
        XCTAssertEqual(first.requester!.userID!, "61644")
        XCTAssertEqual(first.approver!, first.requester!)
    }
}
